block-producer/src/block_builder/prover/mod.rs

```diff
@@ -100,17 +103,62 @@ proc.compute_note_root
     # => [ROOT_{n-1}]
 end
 
-# Stack: [<account root inputs>, <note root inputs>]
-proc.main.2
+#! Compute the chain MMR root
+#! 
+#! Stack: [NUM_LEAVES, num_peaks, PEAK_0, ..., PEAK_{n-1}, PREV_BLOCK_HASH_TO_INSERT]
+#! Output: [CHAIN_MMR_ROOT]
+proc.compute_chain_mmr_root
```

Comment:
I would probably do this slightly differently. Specifically, we could just provide `PREV_CHAIN_MMR` and `PREV_BLOCK_HASH` as inputs, and then use `mmr::unpack`, `mmr::add`, and `mmr::pack` procedures to perform the update (rather than doing the `unpack` manually).

This will require initializing the advice map before we execute this program - but I think we'll need to go there eventually anyway.

Reply:
done! Only tests remaining now